From 8cc99986ff30207054f0eff96562a04f362c4fea Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:36:12 -0400
Subject: [alsa] hda: Make a SND_HDA_PIN_QUIRK macro

Message-id: <140972823573-351-patchset-jkysela@redhat.com>
Patchwork-id: 91047
O-Subject: [RHEL 7.1 PATCH 350/442] ALSA: hda - Make a SND_HDA_PIN_QUIRK macro
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This is cosmetical - it makes the new pin quirk table look better.

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: a2d2fa02b2ec87609295a5bf2391a52e6e7ae111)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_local.h b/sound/pci/hda/hda_local.h
index ebd1fa6..4e2d486 100644
--- a/sound/pci/hda/hda_local.h
+++ b/sound/pci/hda/hda_local.h
@@ -417,6 +417,27 @@ struct snd_hda_pin_quirk {
  int value;   /* quirk value */
 };
 
+#ifdef CONFIG_SND_DEBUG_VERBOSE
+
+#define SND_HDA_PIN_QUIRK(_codec, _subvendor, _name, _value, _pins...) \
+ { .codec = _codec,\
+   .subvendor = _subvendor,\
+   .name = _name,\
+   .value = _value,\
+   .pins = (const struct hda_pintbl[]) { _pins } \
+ }
+#else
+
+#define SND_HDA_PIN_QUIRK(_codec, _subvendor, _name, _value, _pins...) \
+ { .codec = _codec,\
+   .subvendor = _subvendor,\
+   .value = _value,\
+   .pins = (const struct hda_pintbl[]) { _pins } \
+ }
+
+#endif
+
+
 /* fixup types */
 enum {
  HDA_FIXUP_INVALID,
-- 
1.7.1